src/index.ts

```diff
@@ -269,62 +285,86 @@ export default class IlpPluginAsymServer extends MiniAccountsPlugin {
   async _connect (address: string, btpData: BtpData) {
     const { requestId, data } = btpData
     const account = this._getAccount(address)
-    await account.connect()
 
-    const existingChannel = account.getChannel()
-    if (existingChannel) {
+    if (account.getState() === ReadyState.INITIAL) {
+      await account.connect()
+    }
+
+    if (account.isBlocked()) {
+      throw new Error('cannot connect to blocked account. ' +
+        'reconfigure your uplink to connect with a new payment channel.')
+    }
+
+    if (account.getState() > ReadyState.PREPARING_CHANNEL) {
       try {
         this._validatePaychanDetails(account.getPaychan())
-        this._channelToAccount.set(existingChannel, account)
+        this._channelToAccount.set(account.getChannel(), account)
+        await this._watcher.watch(account.getChannel())
         await this._registerAutoClaim(account)
       } catch (e) {
-        this._log.debug('deleting channel because of failed validate. error=', e)
+        this._log.debug('deleting channel because of failed validate.' +
+          ' account=' + account.getAccount() +
+          ' channel=' + account.getChannel() +
+          ' error=', e)
         try {
           await this._channelClaim(account)
           account.deleteChannel()
         } catch (err) {
           this._log.error('could not delete channel. error=', err)
-          // should the account be blocked?
         }
+        this._log.trace('blocking account. account=' + account.getAccount())
+        account.block()
       }
     }
 
     return null
   }
 
   async _fundOutgoingChannel (account: Account, primary: Protocol) {
-    if (account.getClientChannel()) {
+    if (account.getState() !== ReadyState.ESTABLISHING_CLIENT_CHANNEL) {
       this._log.warn('outgoing channel already exists')
```

Comment:
Technically the handleCustomData branch prevents this from being reached, but I want it as an additional safeguard in case the code changes. I did change this to only return the created channel if it's in the READY state, and if it's neither READY nor ESTABLISHING_CLIENT_CHANNEL it throws an error